android/vpnClient/src/main/java/org/mozilla/firefox/vpn/qt/VPNActivity.java

```diff
@@ -188,7 +188,14 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
       // At this point, the user has made a selection on the system config permission modal to either allow or not allow
       // the vpn configuration to be created, so it is safe to run activation retries via ConnectionManager::startHandshakeTimer()
       // without the possibility or re-prompting (flickering) the modal while it is currently being displayed  
-      onServiceMessage(EVENT_VPN_CONFIG_PERMISSION_RESPONSE,"");
+      String result;
+      if(resultCode == RESULT_OK) {
+        result = "granted";
+      }
+      else {
+        result = "denied";
+      }
+      onServiceMessage(EVENT_VPN_CONFIG_PERMISSION_RESPONSE,result);
```

Comment:
@strseb is there a more java/android way to do this?

Reply:
LOL, "Java does not provide this API". I think it does and the documentation is just not updated... I asked the Glean team.

Reply:
Glean team response: "You should be able to call Kotlin code from Java" 🤷‍♀️

Reply:
Technically sure you could import the glean lib into the binding gradle project, just call the telemetry thing from there. 
However i like this approach more, as this keeps VPNActivity with only one job: be small jni glue class. We can handle incoming events using qobjects.